infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -133,37 +139,55 @@
+                log.warn("initializationTask: Management Server waiting to be bootstrapped");
                 Sleep.MILLISECONDS.sleepRecoverably(CHECK_BOOTSTRAP_INTERVAL.toMillis());
             }
+        } catch (InterruptedException e) {
+            // Due to shutdown, which interrupts this thread
+            log.debug("initializationTask: Initialization task interrupted without " +
+                    "management server bootstrapped");
+            return;
+        }
 
-            if (shutdown){
-                log.info("Shutdown signal. Interrupt initialization");
+        // Retry layout recovery if need until success.
+        long recoveryAttempts = 0;
+        while (!shutdown && !recovered) {
+            try {
+                Layout layout = serverContext.copyManagementLayout();
+                if (runRecoveryReconfiguration(layout, getCorfuRuntime())) {
+                    // If recovery succeeds, reconfiguration was successful.
+                    // Save the latest management layout.
+                    serverContext.saveManagementLayout(getCorfuRuntime().getLayoutView().getLayout());
+                    log.info("initializationTask: Recovery completed");
+                    recovered = true;
+                    continue;
+                }
+
+                log.error("initializationTask: Recovery failed {} times. Retrying in {}s.",
+                        ++recoveryAttempts, RECOVERY_RETRY_INTERVAL);
+                Sleep.MILLISECONDS.sleepRecoverably(RECOVERY_RETRY_INTERVAL.toMillis());
+            } catch (InterruptedException e) {
+                // Due to shutdown, which interrupts this thread
+                log.debug("initializationTask: Initialization task interrupted without being recovered");
                 return;
+            } catch (Exception e) {
+                // Retry recovering for any exception encountered.
+                log.error("initializationTask: exception happened during layout recovery, {}", e);
             }
+        }
 
-            // Recover if flag is false
-            if (!recovered) {
-                RecoveryHandler.retryUntilRecovery(serverContext, getCorfuRuntime());
-            }
-
-            this.localMonitoringService.start(METRICS_POLL_INTERVAL);
-            this.remoteMonitoringService.start(POLICY_EXECUTE_INTERVAL);
-
-        } catch (InterruptedException e) {
-            log.error("initializationTask: InitializationTask interrupted.");
-            throw new UnrecoverableCorfuInterruptedError(e);
-        } catch (Exception e) {
-            log.error("initializationTask: Error in initializationTask.", e);
-            throw new UnrecoverableCorfuError(e);
+        // Start monitoring services that deals with failure and healing detection.
+        if (!shutdown) {
+            localMonitoringService.start(METRICS_POLL_INTERVAL);
+            remoteMonitoringService.start(METRICS_POLL_INTERVAL);
```

Comment:
Is this changed from `POLICY_EXECUTE_INTERVAL` to `METRICS_POLL_INTERVAL` on purpose?

Reply:
Looks like a mistake fixing in #1845